sql/sql_type.h

```diff
@@ -722,7 +722,7 @@ class Sec6
   size_t to_string(char *to, size_t nbytes) const
   {
     return m_usec ?
-      my_snprintf(to, nbytes, "%s%llu.%06lu",
+      my_snprintf(to, nbytes, "%s%llu.%06u",
```

Comment:
Unsigned `999_999` requires 20 bits, which requires `long` or above for [Standard C/C++](https://en.cppreference.com/w/cpp/language/types#Properties).
Of course, it could’ve directly chose `uint32_t` (or a variant).

As for the `(uint)` cast… 🧐 that’s a potential bug. `int`s technically (although unorthodox) only have 16 bits (max `65_535`) standardized.

Reply:
I’ve pushed the fix in backport #3485 along with a rebase.